chatlearn/schedule/model_manager.py

```diff
@@ -338,6 +340,11 @@ def _get_model_replica_from_pack(gpu_index, model_pack):
             num_gpus = 1.0 / len(replicas)
             group = i // self.resouce_manager.gpu_per_node
             for replica in replicas:
+                if isinstance(replica.model, VLLMModuleV2) and replica.vllm_engine is None:
+                    num_gpus = num_gpus / 2
```

Comment:
`num_gpus` might be an odd number (1 for example). Consequently, a float would be produced here.

Reply:
num_gpus is float here <=1, since we may colocate models, num_gpus = 1.0 / num_colocated_models.